infinite/index.ts

```diff
@@ -11,18 +11,18 @@ import useSWR, {
   Middleware
 } from 'swr'
 import { useIsomorphicLayoutEffect } from '../src/utils/env'
-import { serialize } from '../src/utils/serialize'
+import { serialize as serializeKey } from '../src/utils/serialize'
 import { isUndefined, UNDEFINED } from '../src/utils/helper'
 import { withMiddleware } from '../src/utils/with-middleware'
 import { SWRInfiniteConfiguration, SWRInfiniteResponse } from './types'
 
 const INFINITE_PREFIX = '$inf$'
 
 const getFirstPageKey = (getKey: KeyLoader<any>) => {
-  return serialize(getKey ? getKey(0, null) : null)[0]
+  return serializeKey(getKey ? getKey(0, null) : null)[0]
 }
 
-export const getInfiniteKey = (getKey: KeyLoader<any>) => {
+export const serialize = (getKey: KeyLoader<any>) => {
```

Comment:
Do you think we should return an array here containing all the custom prefixed keys like `$ctx`, `$len` stuff?
That would be easier to access any of them

Reply:
Not for now. I'm actually worried about exposing too many implementation details, we should do enough research about use cases (e.g. why do they need to access the infinite context?) before thinking about exposing those.

Reply:
I feel the different return type is also bit confusing for new uers (personally), but it's safe to carry on. We can revisit it later.

Would be nice if explicit return type is declared here

Reply:
Oh that's a good point. We should probably only return the data key from `serialize`.

Reply:
unstable_serialize?